pkg/generators/models/testdata/cases/oneof_mapped_discriminator/expected/model_error.go

```diff
@@ -0,0 +1,137 @@
+// This file is auto-generated, DO NOT EDIT.
+//
+// Source:
+//     Title: Test oneOf discriminator support
+//     Version: 0.1.0
+package generatortest
+
+import (
+	"encoding/json"
+	"fmt"
+	validation "github.com/go-ozzo/ozzo-validation/v4"
+	"github.com/mitchellh/mapstructure"
+)
+
+type errorer interface {
+	GetKind() string
+	Validate() error
+}
+
+// Error is a oneOf type.
+type Error struct {
+	data errorer
+}
+
+// MarshalJSON implements the json.Marshaller interface
+func (m Error) MarshalJSON() ([]byte, error) {
+	return json.Marshal(m.data)
+}
+
+// UnmarshalJSON implements the json.Unmarshaller interface
+func (m *Error) UnmarshalJSON(bs []byte) error {
+	discriminator := struct {
+		value string `json:"kind"`
+	}{}
+	err := json.Unmarshal(bs, &discriminator)
+	if err != nil {
+		return err
+	}
+
+	if discriminator.value == "" {
+		return validation.Errors{
+			"kind": fmt.Errorf("can not be empty"),
+		}.Filter()
+	}
+
+	switch discriminator.value {
```

Comment:
we could also choose to use a type switch on the `m.data` to be a little more safe here